client/client.go

```diff
@@ -57,10 +65,34 @@ func (c *Client) CreateDirectChannel(counterparty types.Address, appDefinition t
 	// Pass in a fresh, dedicated go channel to communicate the response:
 	apiEvent := engine.APIEvent{
 		ObjectiveToSpawn: objective,
-		Response:         make(chan engine.Response)}
-
+	}
 	// Send the event to the engine
 	c.engine.FromAPI <- apiEvent
-	// Return the go channel where the response will be sent.
-	return apiEvent.Response
+
+	clientResponse := make(chan ChannelReadyEvent)
+
+	// Starts a go function that listens for our objective to be completed and then dispatchs events to different client channels
+	go func() {
```

Comment:
```suggestion
	// Starts a go routine that listens for our objective to be completed and then dispatches events to different client channels
	go func() {
```

Reply:
I think the use of an anonymous function for the go routine is fine, but my own preference is to name this function so we can document it and get an idea of what its responsibilities are?

That would also make it easier to share with other API methods (we only have one at the moment).

Reply:
This has been refactored in [10c3052](https://github.com/statechannels/go-nitro/pull/208/commits/10c30527ea28ff4c640ea5f090f9657454d99f9b) and now lives in it's own function.